### PR TITLE
nvidia_x11: libPath: add libgbm/libGL libraries

### PR DESCRIPTION
**Pull request by colemickens**
###### Motivation for this change

Fixes up a couple more missing libraries in some of the nvidia library so files.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/nvidia-x11/generic.nix [hunk 1]
```diff
@@ -40,8 +40,11 @@ let
   pkgSuffix = optionalString (versionOlder version "304") "-pkg0";
   i686bundled = versionAtLeast version "391" && !disable32Bit;
 
-  libPathFor = pkgs: pkgs.lib.makeLibraryPath [ pkgs.libdrm pkgs.xorg.libXext pkgs.xorg.libX11
-    pkgs.xorg.libXv pkgs.xorg.libXrandr pkgs.xorg.libxcb pkgs.zlib pkgs.stdenv.cc.cc pkgs.wayland ];
+  libPathFor = pkgs: pkgs.lib.makeLibraryPath ([
+    pkgs.libdrm pkgs.xorg.libXext pkgs.xorg.libX11
+    pkgs.xorg.libXv pkgs.xorg.libXrandr pkgs.xorg.libxcb pkgs.zlib pkgs.stdenv.cc.cc
+    pkgs.wayland pkgs.mesa
+  ] ++ (if useGLVND then [ pkgs.libglvnd ] else [ pkgs.libGL ]));
```
Comment (vcunat) [inline pkgs/os-specific/linux/nvidia-x11/generic.nix:47]: Just to be sure, you are aware that `pkgs.libGL` is just libglvnd stubs on Linux, right?  Doing the dependency directly improves something here, I assume?

Comment (SuperSandro2000) [reply pkgs/os-specific/linux/nvidia-x11/generic.nix:47]: So libGL is for easier support both linux and darwin? I am confused.

Comment (colemickens) [reply pkgs/os-specific/linux/nvidia-x11/generic.nix:47]: @vcunat are you telling me that pkgs.libGL is an alias to libglvnd on Linux or that the libGL source devolves to just being libglvnd itself on Linux or am I not understanding at all?

Comment (colemickens) [reply pkgs/os-specific/linux/nvidia-x11/generic.nix:47]: Okay, I read .../mesa/stubs.nix and I get it now.

I see why this is unnecessary and what useGLVND is needed for insider buiilder.sh.

Thanks for the feedback, this had been a small blind spot until now. I'll fix this up and update tomorrow.

Comment (vcunat) [reply pkgs/os-specific/linux/nvidia-x11/generic.nix:47]: `libGL` provides building against GL libs, on any platform.  Right now, Linux ones use glvnd and Darwin ones do not.

**Comment by colemickens** [comment]
Thanks again @vcunat, this should be better now.

File: pkgs/os-specific/linux/nvidia-x11/generic.nix [hunk 2]
```diff
@@ -40,8 +40,11 @@ let
   pkgSuffix = optionalString (versionOlder version "304") "-pkg0";
   i686bundled = versionAtLeast version "391" && !disable32Bit;
 
-  libPathFor = pkgs: pkgs.lib.makeLibraryPath [ pkgs.libdrm pkgs.xorg.libXext pkgs.xorg.libX11
-    pkgs.xorg.libXv pkgs.xorg.libXrandr pkgs.xorg.libxcb pkgs.zlib pkgs.stdenv.cc.cc pkgs.wayland ];
+  libPathFor = pkgs: pkgs.lib.makeLibraryPath ([
+    pkgs.libdrm pkgs.xorg.libXext pkgs.xorg.libX11
+    pkgs.xorg.libXv pkgs.xorg.libXrandr pkgs.xorg.libxcb pkgs.zlib pkgs.stdenv.cc.cc
+    pkgs.wayland pkgs.mesa pkgs.libGL
+  ]);
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/nvidia-x11/generic.nix:47]: Can we do a with pkgs here?

**Review by fabianhjr — APPROVED** [review]
Makes steam and webgl usable under wayland and nvidia on NixOS.

Tested alongside #145319 and #138180

**Review by jonringer — APPROVED** [review]
diff LGTM

**Pull request merged**